platform/viewer/src/appExtensions/GenericViewerCommands/commandsModule.js

```diff
@@ -16,24 +12,63 @@ const actions = {
     newIndex = newIndex < 0 ? maxIndex : newIndex;
 
     store.dispatch(setViewportActive(newIndex));
-  }
+  },
+  updateViewportDisplaySet: ({ viewports, direction }) => {
+    const viewportSpecificData = { ...viewports.viewportSpecificData };
+    const activeViewport = viewportSpecificData[viewports.activeViewportIndex];
+    const studyMetadata = utils.studyMetadataManager.get(
+      activeViewport.studyInstanceUid
+    );
+
+    if (!studyMetadata) {
+      return;
```

Comment:
@dannyrb we could create commands to wrap up state update logic which dispatches or a new action/reducer (redux) with a connected component? whats the model we should strive in this case?

Reply:
I'm not sure I follow. Can you pseudo code an example?